File: 프로그래머스/2/42586. 기능개발/기능개발.py
# 반복문을 progress 가 어떤 원소가 있을때까지 계속 돌린다. 
# 0번재 값이 100이 되면 뒤의 원소가 100이 되었는지 확인한다. 
# 100인 거는 다 꺼낸다. 
# 꺼낸 갯수 만큼 return 의 리스트에 append 한다.

def solution(progresses, speeds):
    answer = []
    while progresses:
        count=0
        if progresses[0]>=100:
            while progresses[0]>=100:
                progresses.pop(0)
                speeds.pop(0)
                count+=1
                if progresses==[]:
                    break
            answer.append(count)
            
        if progresses==[]:
            break
        for i in range(len(progresses)):
            progresses[i]+=speeds[i]
            
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answer = new ArrayList<>();
        while(progresses.length!=0){
            int count=0;
            if(progresses[0]>=100){
                while(progresses[0]>=100){
                    progresses=pop(progresses);
                    speeds=pop(speeds);
                    count+=1;
                    if(progresses.length==0){
                        break;
                    }
                }
                answer.add(count);
            }
            
            if(progresses.length==0){
                break;
            }
            for(int i=0;i<progresses.length;i++){
                progresses[i]+=speeds[i];
            }
        }
        int[] intArray = convertArrayListToIntArray(answer);

        return intArray;
    }
    public int[] pop(int[] arr){
        ArrayDeque<Integer> newarr = new ArrayDeque<>();
        for(int i:arr){
            newarr.add(i);
        }
        newarr.pollFirst();
        int[] array = new int[newarr.size()];
        int count=0;
        for(int i:newarr){
            array[count]=i;
            count++;
        }
        return array;
    }
    public static int[] convertArrayListToIntArray(ArrayList<Integer> arrayList) {
        int[] intArray = new int[arrayList.size()];
        for (int i = 0; i < arrayList.size(); i++) {
            intArray[i] = arrayList.get(i);
        }
        return intArray;
    }
}